### DISCUSSION TITLE
Table - Search and Filter

### DISCUSSION THREAD

**Discussion by emrecoban** [rsuite/rsuite#3282] [Q&A] [answered]
How would one go about building a filter by row or search? Any sample code?

**Updated:**
I decided to use the `.filter()` method with the `useEffect` hook.

A sample:
![image](https://github.com/rsuite/rsuite/assets/4332481/3ed852e2-1030-466b-bb21-d36547dc1a5a)

**Comment by SevenOutman** [comment, marked as answer]
Table component does not handle filtering. You could filter your data before passing them into Table.

Here's an example:

```jsx
const [filterText, setFilterText] = useState('')

const filteredData = useMemo(() => {
  if (!filterText) return allData
  return filter(allData, filterText)
}, [allData, filterText])

return (
  <>
    <Input value={filterText} onChange={value => setFilterText(value)} />
    <Table data={filteredData} />
  </>
)
```

**Reply by emrecoban** [reply]
Yes, I use this method. Thanks.
